packages/core/src/api/applicationCommands.ts

```diff
@@ -0,0 +1,234 @@
+
+	/**
+	 * Fetches all global commands for a application
+	 *
+	 * @param applicationId - The application id to fetch commands for
+	 * @param options - The options to use when fetching commands
+	 */
+	public async getGlobalCommands(applicationId: Snowflake, options: { with_localizations?: boolean } = {}) {
+		return this.rest.get(Routes.applicationCommands(applicationId), {
+			query: makeURLSearchParams(options),
+		}) as Promise<RESTGetAPIApplicationCommandsResult>;
+	}
+
+	/**
+	 * Creates a new global command
+	 *
+	 * @param applicationId - The application id to create the command for
+	 * @param data - The data to use when creating the command
+	 */
+	public async createGlobalCommand(applicationId: Snowflake, data: RESTPostAPIApplicationCommandsJSONBody) {
+		return this.rest.post(Routes.applicationCommands(applicationId), {
+			body: data,
+		}) as Promise<RESTPostAPIApplicationCommandsResult>;
+	}
+
+	/**
+	 * Fetches a global command
+	 *
+	 * @param applicationId - The application id to fetch the command from
+	 * @param commandId - The command id to fetch
+	 */
+	public async getGlobalCommand(applicationId: Snowflake, commandId: Snowflake) {
+		return this.rest.get(
+			Routes.applicationCommand(applicationId, commandId),
+		) as Promise<RESTGetAPIApplicationCommandResult>;
+	}
+
+	/**
+	 * Edits a global command
+	 *
+	 * @param applicationId - The application id of the command
+	 * @param commandId - The id of the command to edit
+	 * @param data - The data to use when editing the command
+	 */
+	public async editGlobalCommand(
+		applicationId: Snowflake,
+		commandId: Snowflake,
+		data: RESTPatchAPIApplicationCommandJSONBody,
+	) {
+		return this.rest.patch(Routes.applicationCommand(applicationId, commandId), {
+			body: data,
+		}) as Promise<RESTPatchAPIApplicationCommandResult>;
+	}
+
+	/**
+	 * Deletes a global command
+	 *
+	 * @param applicationId - The application id of the command
+	 * @param commandId - The id of the command to delete
+	 */
+	public async deleteGlobalCommand(applicationId: Snowflake, commandId: Snowflake) {
+		await this.rest.delete(Routes.applicationCommand(applicationId, commandId));
+	}
+
+	/**
+	 * Overwrites global commands
+	 *
+	 * @param applicationId - The application id to overwrite commands for
+	 * @param data - The data to use when overwriting commands
+	 */
+	public async bulkOverwriteGlobalCommands(applicationId: Snowflake, data: RESTPutAPIApplicationCommandsJSONBody) {
+		return this.rest.put(Routes.applicationCommands(applicationId), {
+			body: data,
+		}) as Promise<RESTPutAPIApplicationCommandsResult>;
+	}
+
+	/**
+	 * Fetches all commands for a guild
+	 *
+	 * @param applicationId - The application id to fetch commands for
+	 * @param guildId - The guild id to fetch commands for
+	 * @param data - The data to use when fetching commands
+	 */
+	public async getGuildCommands(
+		applicationId: Snowflake,
+		guildId: Snowflake,
+		data: { with_localizations?: boolean } = {},
```

Comment:
Isn't there a discord-apoli-types for this?

Reply:
There is now, needs a discord-api-types dependency bump
`RESTGetAPIApplicationGuildCommandsQuery`

Reply:
I'll put a TODO here and fix it when -types is bumped in main.